src/datetime/mod.rs

```diff
@@ -79,11 +79,141 @@ pub enum SecondsFormat {
+/// instead of a localized time.
+///
+/// # Timezones
+///
+/// [`DateTime`] is generic over types implementing [`TimeZone`].
+///
+/// [`TimeZone`] implementations differ in the amount of information they have available, and
+/// because of that the range of operations they can do correctly.
+///
+/// Ordered from most to least information:
+/// - Types implementing proper timezone support using the IANA time zone database, such as in the
+///   [`chrono_tz`](https://crates.io/crates/chrono_tz`) and [`tzfile`](
+///   https://crates.io/crates/tzfile) crates. They support changes like offset shifts and daylight
+///   saving time.
+/// - [`Local`] matches the current timezone settings of the operating system. That may however be a
+///   simplified implementation, not describing all complexities like the timezone database does.
+/// - [`FixedOffset`] stores only the offset from UTC of a specific datetime, without knowing the
+///   timezone. This is the preferred type for an end result, a timestamp.
+/// - [`Utc`] is the universal time, carrying no information about a local time at all. It is
+///   continuous, without gaps or ambiguities.
+///
+/// # Creating a `DateTime`
+///
+/// ## Combine from primitive types
+/// Methods are available on multiple types for convenience.
+/// - Combine a [`TimeZone`] and a _local_ [`NaiveDateTime`]:
+///   - [`NaiveDateTime::and_local_timezone`]
+///   - [`TimeZone::from_local_datetime`]
+///   - [`DateTime<Tz: TimeZone>::from_local`]
+/// - Combine a [`TimeZone`] and a _UTC_ [`NaiveDateTime`]:
+///   - [`TimeZone::from_utc_datetime`]
+///   - [`DateTime<Tz: TimeZone>::from_utc`]
+/// - (convenience) Combine a [`TimeZone`] with _local_ year, month, day and time components:
+///   - [`TimeZone::with_ymd_and_hms`]
+///
+/// ## From the current time
+/// (Requires the default `clock` feature.)
+/// - [`Local.now()`](offset/struct.Local.html#method.now)
+/// - [`Utc.now()`](offset/struct.Utc.html#method.now)
+///
+/// ## Parsing
+/// - [`parse_from_str`](#method.parse_from_str) for parsing with a format string.
+/// - [`parse_from_rfc2822`](#method.parse_from_rfc2822) for parsing a string such as
+///   `Tue, 1 Jul 2003 10:52:37 +0200` to a [`DateTime<FixedOffset>`].
+/// - [`parse_from_rfc3339`](#method.parse_from_rfc3339) for parsing a string such as
+///   `2003-07-01T10:52:37+02:00` to a [`DateTime<FixedOffset>`].
+/// - [`TimeZone::datetime_from_str`] parsing from a string without timezone information
+///   (or with an offset that must match that of the timezone).
+///
+/// ## From common external types:
+/// - Combine a [`TimeZone`] with a UTC Unix timestamp:
+///   - [`TimeZone::timestamp_opt`]
+///   - [`TimeZone::timestamp_millis_opt`]
+///   - [`TimeZone::timestamp_nanos`]
+/// - From the standard library [`SystemTime`](
+///   https://doc.rust-lang.org/std/time/struct.SystemTime.html) (requires the default `std`
```

Comment:
I prefer separate links. The raw source docs have cleaner widths and are easier to read/verify.  Also, std library things don't need leading `http://doc....`.

```rust
/// From the standard library [`SystemTime`]:
///
/// [`SystemTime`]: std::time::SystemTime
```